### ISSUE TITLE
Improve quality of Spoon's API documentation

### ISSUE THREAD

**Issue by slarse** [INRIA/spoon#3923]
A lot of the documentation of the public API is missing documentation for things like parameters and return types (think `@param` and `@return` tags). In my opinion, the most glaring omissions of these details are in the metamodel docs, found in the subpackages of [`spoon.reflect`](https://github.com/INRIA/spoon/tree/master/src/main/java/spoon/reflect).

For example, at the time of writing this, [CtImport](https://github.com/INRIA/spoon/blob/009ad17627741c67d1a04e1f363730e6dfa17ee3/src/main/java/spoon/reflect/declaration/CtImport.java) is entirely missing tags in its Javadocs.

I will expand on this issue with more instructions soon.

**Comment by bhargavi-kotha** [comment]
@slarse Is this still open to start with contribution? And i'm willing to work on GSoC 2022 Implement Javadoc quality project idea.
can u please guide me on how to start with if not this PR to go on. I have experience on java based microservices and spring boot.I have implemented unit test cases with sonar coverage 85% in many of the projects.

**Comment by MartinWitt** [comment]
Hi @bhargavi-kotha,

Yes, this issue is still open and there should be countless Javadoc issues left. You are pretty fast because I have still no confirmation that we are accepted as GSoC project. In the next days(when I/we get the confirmation) we create a new GSoC post on things you can do to get into the project spoon. For now, you can have a look at the old post from @monperrus https://github.com/INRIA/spoon/issues/3828 from 2021.

**Comment by slarse** [comment]
What @MartinWitt said :)

**Comment by bhargavi-kotha** [comment]
Thanks @MartinWitt and @slarse . I would go through any documentation and explore the project. would start off with any issue, until we get confirmation.

**Comment by monperrus** [comment]
Javadoc warning's today

```
[WARNING] Javadoc Warnings
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/compiler/Environment.java:579: warning - invalid usage of tag >
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/processing/AbstractParallelProcessor.java:34: warning - Tag @link: reference not found: Executors#newFixedThreadPool()
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/refactoring/Refactoring.java:234: warning - @param argument "input" is not a parameter name.
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/code/CtTypePattern.java:29: warning - invalid usage of tag >
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/factory/Factory.java:371: warning - Tag @see: reference not found: CodeFactory#createJavaDocTag()
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/factory/Factory.java:371: warning - Tag @see: reference not found: CodeFactory#createJavaDocTag()
[WARNING] javadoc: warning - Tag @link: reference not found: #toString()
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/visitor/PrettyPrinter.java:73: warning - Tag @link: reference not found: #toString()
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/visitor/PrettyPrinter.java:73: warning - Tag @link: reference not found: #toString()
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/visitor/PrettyPrinter.java:73: warning - Tag @link: reference not found: #toString()
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/compiler/Environment.java:579: warning - invalid usage of tag >
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/support/template/Parameters.java:221: warning - @param argument "targetType" is not a parameter name.
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/support/util/internal/ElementNameMap.java:28: warning - invalid usage of tag >
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/visitor/PrettyPrinter.java:73: warning - Tag @link: reference not found: #toString()
[WARNING] /home/martin/martin-no-backup/spoon/spoon-core/src/main/java/spoon/reflect/visitor/PrettyPrinter.java:73: warning - Tag @link: reference not found: #toString()

```

**Comment by rishivijayv** [comment]
For new contributors, I faced some roadblocks today trying to get the (_incredibly useful!_) `check-javadoc-regressions.py` script up and running. 

When I ran the `check-javadoc-regressions.py` script, it made a call to the `run-with-spoon-javadoc-classpath.sh`, which in turn ran the `mvn dependency:build-classpath` command from within the `spoon-javadoc` directory and [seems to direct the output to `/dev/null`](https://github.com/INRIA/spoon/blob/master/chore/run-with-spoon-javadoc-classpath.sh#L8). I was getting errors when running the command but the redirection made it difficult to figure out why.

I then ran `mvn dependency:build-classpath` from within `spoon-javadoc` manually, and eventually realized that it was likely because maven was having trouble locating my `spoon-core` snapshot jar in the local maven repository. I then ran `mvn install` from the root directory of the project, after this, I maven then had trouble locating the `spoon-pom` jar. (_PS: To see where your local maven repository is, you can open the project in IntelliJ, click on the `m` icon on the vertical bar on the right, and click on the `Repositories` option_).

Eventually, what seemed to fix things for me was **running `mvn install` from within the `spoon-pom/` directory** and then using the `check-javadoc-regressions.py` script. Here's a good description of what `mvn install` does (along with a couple other maven plugins): https://stackoverflow.com/a/3660935

This might be common knowledge, but in case someone wasn't aware of this workflow (like me :)), I hope this saves you some time! 

**TL;DR**: Make sure you have run `mvn install` from within `spoon-pom/` before using the `check-javadoc-regressions.py` script

**Comment by AzazelSensei** [comment]
Opened a PR for ProcessingManager: https://github.com/INRIA/spoon/pull/6874
